lib/linguist/generated.rb

```diff
@@ -509,5 +510,12 @@ def generated_grpc_cpp?
       return false unless lines.count > 1
       return lines[0].start_with?("// Generated by the gRPC")
     end
+
+    # Internal: Is this a snap file of Jest?
+    #
+    # Return true or false
+    def generated_jest_snap?
+      name.include?("__snapshots__") && name.end_with?(".snap")
```

Comment:
Could you switch the two predicates? I expect the extension to be a better (and cheaper) discriminator than the path.